src/api/users/getUsers.js

```diff
@@ -8,6 +8,10 @@ type ApiResponseUsers = {|
   members: User[],
 |};
 
+// TODO: If we start to use this, we need to convert `.avatar_url` to
+// an AvatarURL instance, like we do in `registerForEvents` and
+// `EVENT_USER_ADD` and `EVENT_USER_UPDATE`.
```

Comment:
> This might mean it's time to add a `RawUser` and `RawCrossRealmBot` in `modelTypes.js`, since at least the former shows up in several different files. In particular that'd mean that if for some reason we add another conversion in the future, this type annotation would keep meaning the same thing (because it'd refer to `RawUser`, which we'd edit to keep it meaning the same thing) rather than silently start meaning the new conversion.

Hmm, maybe. I think `avatar_url: string | null` would be correct for `GET /users` (this binding, which we don't currently use) and `GET /users/{user_id}`, which we don't have a binding for.

But it wouldn't be correct for the `/register` response, which we do use. There, `avatar_url` might be missing, for a couple of reasons that only apply to the `/register` response, like `user_avatar_url_field_optional` and zulip/zulip@58ee3fa8c. (Which I guess means I should spell it as `avatar_url?: string | null` in `RawInitialDataRealmUser` in `initialDataTypes`, instead of `avatar_url: string | void | null`?)

I'm not sure how to address the differences in how user/cross-realm bot objects come to us from the server, but I felt OK making `RawInitialDataRealmUser` because it only covers one of those ways (the `/register` response). Making one `RawUser` or one `RawCrossRealmBot` in `modelTypes.js` would probably be OK if we don't mind it representing a union of the raw-data shapes we get from everywhere.

Reply:
> But it wouldn't be correct for the `/register` response, which we do use. There, `avatar_url` might be missing (not `null` or a string), for a couple of reasons that only apply to the `/register` response,

Hmm, I see. Yeah, that makes it awkward to try to have a common type.



> (Which I guess means I should spell it as `avatar_url?: string | null` in `RawInitialDataRealmUser` in `initialDataTypes`, instead of `avatar_url: string | void | null`?)

Yep! In fact as a general rule, if some type is supposed to come from decoding some JSON, then there definitely shouldn't be `void` in it -- JSON has nothing corresponding to JS's `undefined`. The closest equivalents to a JS `{ foo: undefined }` in JSON are `{ "foo": null }` or just `{}`.